### ISSUE TITLE
Bug: `sort-classes`: Static properties referenced using MyClass.x are not detected as dependencies

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#237]
### Describe the bug

Properties dependent on other properties through the `this.x` keyword are properly detected and sorted. However, static properties dependent on other static properties through the `MyClass.x` keyword are not.

### Code example

The following code is invalid and will try to put `a` before `b` when it should not.

```typescript
class Class {
  static b = 1

  static a = Class.b
}
```

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

v3.3.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Fixed in e7c113d.
Released in [v3.4.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.4.0).
